src/core/execute.c

```diff
@@ -463,6 +469,10 @@ int exec_spawn(
                         environ,
                         cg_unified() > 0 ? subcgroup_path : NULL,
                         &pidref);
+
+        if (unit->manager->starting_ambient_set != 0 && MANAGER_IS_USER(unit->manager))
+                capability_ambient_set_apply(0, /* also_inherit= */ false);
```

Comment:
this should also have a comment explaining the situation, i.e. that this undoes the ambient stuff we did above